webapp/Connector/src/app/view/module/NonIntegratedDataAvailability.js

```diff
@@ -174,53 +160,152 @@ Ext.define('Connector.view.module.NonIntegratedDataAvailability', {
     hideDataStatusTooltip : function() {
         this.fireEvent('hideTooltip');
     },
+
+    getColTemplate : function() {
+        console.warn('need to override getColTemplate');
+    }
+});
+
+Ext.define('Connector.view.module.StudyNonIntegratedData', {
+
+    xtype: 'app.module.studynonintegrateddata',
+
+    extend: 'Connector.view.module.NonIntegratedDataAvailability',
+
+    cls: 'module learn-data-available-module',
+
+    initComponent : function() {
+        this.callParent();
+    },
+
+    getData : function() {
+        return this.initialConfig.data.model.data.non_integrated_assay_data;
+    },
+
+    getTitleData: function(data) {
+        if (data.non_integrated_assay_data.length > 0 && data.non_integrated_assay_data_has_permission) {
+            data.hasDetails = true;
+        }
+        else {
+            data.hasDetails = false;
+        }
+        return data;
+    },
+
     getColTemplate : function() {
-        var me = this;
         return new Ext4.XTemplate(
-           '<tpl>',
+                '<tpl>',
                 // '<table><tr><td>',
 
                 // case when there is both a link to the assay learn page and data to download
                 '<tpl if="isLinkValid && hasAssayLearn">',
                 '<ul class="non-integrated-data-ul">',
-                    '<li class="non-integrated-data-li">',
-                        '<a href="#learn/learn/Assay',
-                        '/{assayIdentifier}">',
-                        '{label:htmlEncode}',
-                        '</a>',
-                        '&nbsp;{suffix}&nbsp;',
-                        '<a href="{filePath}" target="_blank"><img alt="{assayIdentifier}" src="' + LABKEY.contextPath + '/Connector/images/download-icon.svg' + '" height="13" width="13" align="left"/></a>',
-                    '</li>',
+                '<li class="non-integrated-data-li">',
+                '<a href="#learn/learn/Assay',
+                '/{assayIdentifier}">',
+                '{label:htmlEncode}',
+                '</a>',
+                '&nbsp;{suffix}&nbsp;',
+                '<a href="{filePath}" target="_blank"><img alt="{assayIdentifier}" src="' + LABKEY.contextPath + '/Connector/images/download-icon.svg' + '" height="13" width="13" align="left"/></a>',
+                '</li>',
                 '</ul>',
```

Comment:
Looks like indentation here (lines 203-210) and in the other places in this function is lost.

Reply:
ooh weird, IntelliJ must have done that on copy/paste my bad!